server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -317,6 +329,39 @@ public String executor() {
         }
     }
 
+    TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) throws Exception {
+        // Read enum
+        switch (extensionRequest.getRequestType()) {
+            case REQUEST_EXTENSION_CLUSTER_STATE:
+                ClusterStateResponse clusterStateResponse = new ClusterStateResponse(
+                    clusterService.getClusterName(),
+                    clusterService.state(),
+                    false
+                );
+                return clusterStateResponse;
+            case REQUEST_EXTENSION_LOCAL_NODE:
+                LocalNodeResponse localNodeResponse = new LocalNodeResponse(clusterService);
+                return localNodeResponse;
+            case REQUEST_EXTENSION_CLUSTER_SETTINGS:
+                ClusterSettingsResponse clusterSettingsResponse = new ClusterSettingsResponse(clusterService);
+                return clusterSettingsResponse;
+            case REQUEST_EXTENSION_ACTION_LISTENER_ON_FAILURE:
+                return handleExtensionActionListenerOnFailureRequest(extensionRequest.getFailureException());
+            default:
+                throw new Exception("Handler not present for the provided request");
+        }
+    }
+
+    private ExtensionBooleanResponse handleExtensionActionListenerOnFailureRequest(String failureException) throws Exception {
+        try {
+            listener.onFailure(new Exception(failureException));
```

Comment:
Can we pick a more narrow exception to send here?  Not sure which one would apply.  Do we have a custom `OpenSearchException`?

Reply:
We can, but the actual exception type will be in the exception message anyways.

Reply:
I changed it to OpenSearchException